test_proxy/test_proxy.py

```diff
@@ -0,0 +1,187 @@
+        if not request_q.empty():
+            json_data = format_dict(request_q.get())
+            fn_name = json_data.pop("proxy_request")
+            print(f"--- running {fn_name} with {json_data}")
+            out_q = queue_pool[json_data.pop("response_queue_idx")]
+            client_id = json_data.pop("client_id")
+            client = client_map.get(client_id, None)
+            # handle special cases for client creation and deletion
+            if fn_name == "CreateClient":
+                if use_legacy_client:
+                    client = client_handler_legacy.LegacyTestProxyClientHandler(**json_data)
+                else:
+                    client = client_handler.TestProxyClientHandler(**json_data)
+                client_map[client_id] = client
+                out_q.put(True)
+            elif client is None:
+                out_q.put(RuntimeError("client not found"))
+            elif fn_name == "CloseClient":
+                # TODO: link this to actual client closing functionality
+                # current implementation happens at test_proxy level
+                # see TestMutateRows_Generic_CloseClient
+                client.close()
+                out_q.put(True)
+            elif fn_name == "RemoveClient":
+                client_map.pop(client_id, None)
+                out_q.put(True)
+                print("")
+            else:
+                # run actual rpc against client
+                async def _run_fn(out_q, fn, **kwargs):
+                    result = await fn(**kwargs)
+                    out_q.put(result)
+                fn = getattr(client, fn_name)
+                task = asyncio.create_task(_run_fn(out_q, fn, **json_data))
+                await asyncio.sleep(0)
+                background_tasks.add(task)
+                task.add_done_callback(background_tasks.remove)
+        await asyncio.sleep(0.01)
+
+
+def client_handler_process(request_q, queue_pool, legacy_client=False):
+    import asyncio
+    asyncio.run(client_handler_process_async(request_q, queue_pool, legacy_client))
+
+
+import argparse
+p = argparse.ArgumentParser()
+p.add_argument("--port", dest='port', default="50055")
+p.add_argument('--legacy-client', dest='use_legacy', action='store_true', default=False)
+
+if __name__ == "__main__":
+    port = p.parse_args().port
+    use_legacy_client = p.parse_args().use_legacy
+
+    # start and run both processes
+    # larger pools support more concurrent requests
+    response_queue_pool = [multiprocessing.Queue() for _ in range(100)]
+    request_q = multiprocessing.Queue()
+
+    # run client in forground and proxy in background
+    # breakpoints can be attached to client_handler_process
+    proxy = multiprocessing.Process(
+        target=grpc_server_process,
+        args=(
+            request_q,
+            response_queue_pool,
+            port
+        ),
+    )
+    proxy.start()
+    client_handler_process(request_q, response_queue_pool, use_legacy_client)
```

Comment:
How is it terminated? For java client testing, we use this: https://github.com/googleapis/java-bigtable/blob/main/.kokoro/conformance.sh#L59

Reply:
Good point. It currently isn't automatically terminated, but I will need to add that for running automated tests.

I may just do this as part of the kokoro script when I add this to kokoro. Do you think I should make that part of this PR?

Reply:
sg.